### ISSUE TITLE
Button element missing accessible name.

### ISSUE THREAD

**Issue by hilvlmedivh** [hilvlmedivh/initrepo#3]
**Rule description**  
One or more button elements, without an ARIA-assigned role, not intentionally hidden in the DOM and available to assistive technologies, does not have a mechanism that allows an accessible name value to be calculated.

**Tool name**  
Access Engine

**Severity**  
Critical

**WCAG Success criteria**  
[2.5.3 Label in Name - Level A](https://www.w3.org/WAI/WCAG22/Understanding/label-in-name)  
[1.3.1 Info and Relationships - Level A](https://www.w3.org/WAI/WCAG22/Understanding/info-and-relationships)  
[4.1.2 Name, Role, Value - Level A](https://www.w3.org/WAI/WCAG22/Understanding/name-role-value)

**Why it matters**  
When <code>button</code>  elements do not have an accessible name, assistive technologies may incorrectly render the name or provide no name at all to users. When unnamed controls are present, users of assistive technologies may not be able to complete a form.

**What to do**  
Provide an accessible name for all buttons. The text or alternative text of images within buttons is used as the accessible name of the button. When on-screen text is not available, authors can use another method to provide an accessible name, such as by using aria-labelledby, aria-label, or title attribute, to provide the accessible name. The aria-labelledby attribute must contain one or more IDs of elements that label the button. The <code>aria-label</code> attribute is best when there is no on-screen label. Authors should place the string that labels the form field in the <code>aria-label</code> attribute.

Authors can also identify an accessible name using an associated label element whenever an on-screen label is present. The <code>label</code> element's value is then explicitly associated with the relevant form field using the for and id attributes. Use of the <code>label</code> element provides an added benefit as it increases the target area that users can click to focus the form field. In certain instances, however, it is impractical or 'undesirable' to use an explicit label for a button; this is largely driven by the fact that the value between the opening and closing label tags must, by definition, be a block of displayed text.

**Task in the Level Access Platform**  
https://waesqa.hub.dev.dev.levelaccess.io/my-tasks/6a8348d85a680b03ab54646d/6a834c46cffbc80dd64323a1

**Comment by hilvlmedivh** [comment]
Task pushed to GitHub Issues by Markiian Skrypets from Level Access Platform.

Posted by Markiian Skrypets via Level Access

**Comment by hilvlmedivh** [comment]
<a href="https://waesqa.hub.dev.dev.levelaccess.io/tasks/6a834c46cffbc80dd64323a1/attachments/6a834c46cffbc80dd643239f/download" target="_blank" rel="noopener noreferrer">Linked Findings.csv</a>

**Comment by hilvlmedivh** [comment]
Task pushed to GitHub Issues by Markiian Skrypets from Level Access Platform.

Posted by Markiian Skrypets via Level Access
